Yoast/wordpress-seo
# Clean up Spanish stemmer

## Context


* We introduced the Spanish stemmer based on a pre-existing stemmer. We didn’t adopt it to our code standards in the initial iteration since we were going to refactor it anyways. In this PR we adapt the external stemmer to our code standard.

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Improves code quality of the Spanish stemmer and changes the way we stem words ending in -mente.
* [Yoast SEO free] Improves the stemming of words that look like ending in suffix -mente, e.g. "clemente" was incorrectly stemmed to "cl", now it's stemmed to "clement".

## Relevant technical choices:

* Documentation has been added for all functions, so the `require-jsdoc` exception has been removed.
* The eslint checks for `max-statements` and `complexity` are still disabled. Even though smaller functions were split out from the main stem function, that main function is still quite complex. However, only removing `require-jsdoc` and some refactoring, it's already improving the readability of the stemmer.


## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* Run `yarn test` and make sure that everything passes
* Set the browser language to Spanish
* Create a post with at least 300 words
* Add word `clementes` in the text
* Add word `clemente` as the keyphrase
* In `keyphrase density` assessment, the focus keyphrase should be found 1 time


### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

* Set the browser language to Spanish
* Create a post with at least 300 words
* Add word `clementes` in the text
* Add word `clemente` as the keyphrase
* In `keyphrase density` assessment, the focus keyphrase should be found 1 time

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Documentation


## Quality assurance


Fixes https://yoast.atlassian.net/browse/LINGO-229
